src/Resources/config/collector.xml

```diff
@@ -9,6 +9,7 @@
         <service id="csa_guzzle.data_collector.history_bag" class="ArrayObject" />
 
         <service id="csa_guzzle.data_collector.guzzle" class="Csa\Bundle\GuzzleBundle\DataCollector\GuzzleCollector">
+            <argument type="constant">\Csa\Bundle\GuzzleBundle\DataCollector\GuzzleCollector::MAX_BODY_SIZE</argument>
```

Comment:
Maybe use an empty argument here instead, as the default value is actually defined in the configuration class.

The main problem was mainly the `addArgument` call (which actually inverts both arguments, as the second argument didn't exist at the time), which should indeed be replaced by `replaceArgument`, which you actually did.